modules/sharedaddy/sharing.php

```diff
@@ -547,25 +547,48 @@ public function management_page() {
 	}
 }
 
+/**
+ * Callback to get the value for the jetpack_sharing_enabled field.
+ *
+ */
+function jetpack_post_sharing_get_value( array $post ) {
```

Comment:
It may just be me, but I find those function names a bit confusing, as I don't know right away what value this is referring to. It becomes even more confusing with values like `sharing_disabled`, set to true for example, that we negate to say that `jetpack_sharing_enabled` is... false. 😌 

What do you think?

Reply:
Well, that's the confusion I'm trying to eliminate by renaming the post_meta for the API keys. I'm open to suggestions on the function names, I just tried to name all the callbacks according to the same pattern.

Reply:
To save you the context reloading, these options are stored in post_meta on the post that looks like this:

Sharing: sharing_disabled meta key is set only if the global site setting is enabled, and we want to suppress sharing buttons on that particular post. Otherwise not set. It's not possible to enable sharing buttons when they are disabled globally.

Likes: Stored as an override of the global setting for the posts that have been set to invert the global setting. (I know. Terrible.) So if Likes are ON globally, a particular post can have them disabled with the switch_like_status meta; conversely, if Likes are OFF globally, a particular post can still have them enabled.

Reply:
This might be a good candidate to leave a note, saying that this is complicated, and link to public documentation on the behavior. Non-blocking of course @codebykat .

Reply:
Thanks for the additional info. That's helpful. I am not quite sure how to make that easier, but a note may help, as Kerry suggested. 🙇

Reply:
I added some more verbose comments to the function definitions.